src/protocol/invite.rs

```diff
@@ -42,25 +42,38 @@ pub(crate) struct InvitationType(pub libparsec::types::InvitationType);
 
 #[pymethods]
 impl InvitationType {
-    #[new]
-    fn new(invitation_str: &str) -> PyResult<Self> {
-        match invitation_str {
-            "DEVICE" => Ok(Self(libparsec::types::InvitationType::Device)),
-            "USER" => Ok(Self(libparsec::types::InvitationType::User)),
-            _ => Err(PyNotImplementedError::new_err("")),
-        }
-    }
-
-    #[classmethod]
+    #[classattr]
     #[pyo3(name = "DEVICE")]
-    fn device(_cls: &PyType) -> Self {
-        Self(libparsec::types::InvitationType::Device)
+    fn device() -> &'static PyObject {
+        lazy_static::lazy_static! {
+            static ref VALUE: PyObject = {
+                Python::with_gil(|py| {
+                    InvitationType(libparsec::types::InvitationType::Device).into_py(py)
+                })
+            };
+        };
+        &VALUE
     }
 
-    #[classmethod]
+    #[classattr]
     #[pyo3(name = "USER")]
-    fn user(_cls: &PyType) -> Self {
-        Self(libparsec::types::InvitationType::User)
+    fn user() -> &'static PyObject {
+        lazy_static::lazy_static! {
+            static ref VALUE: PyObject = {
+                Python::with_gil(|py| {
+                    InvitationType(libparsec::types::InvitationType::User).into_py(py)
+                })
+            };
+        };
+        &VALUE
+    }
+
+    #[getter]
+    fn value(&self) -> &'static str {
+        match self.0 {
+            libparsec::types::InvitationType::Device => "DEVICE",
+            libparsec::types::InvitationType::User => "USER",
+        }
     }
 
     fn __richcmp__(&self, other: &Self, op: CompareOp) -> PyResult<bool> {
```

Comment:
Can this `__richcmp__` be generated by the `gen_proto` macro? I didn't know this macro existed when I wrote it.
It may look like this

```suggestion
    gen_proto!(InvitationType, __richcmp__, eq);
```

Reply:
done.